src/Symfony/Component/Validator/Constraints/MacAddressValidator.php

```diff
@@ -23,6 +23,71 @@
  */
 class MacAddressValidator extends ConstraintValidator
 {
+    /**
+     * Checks whether an MAC address is valid.
+     */
+    private static function checkMac(string $mac, mixed $type): bool
+    {
+        if (!filter_var($mac, \FILTER_VALIDATE_MAC)) {
+            return false;
+        }
+
+        return match ($type) {
+            MacAddress::LOCAL_ALL => self::isLocal($mac),
+            MacAddress::LOCAL_NO_BROADCAST => self::isLocal($mac) && !self::isBroadcast($mac),
+            MacAddress::LOCAL_UNICAST => self::isLocal($mac) && self::isUnicast($mac),
+            MacAddress::LOCAL_MULTICAST => self::isLocal($mac) && !self::isUnicast($mac),
+            MacAddress::LOCAL_MULTICAST_NO_BROADCAST => self::isLocal($mac) && !self::isUnicast($mac) && !self::isBroadcast($mac),
+            MacAddress::UNIVERSAL_ALL => !self::isLocal($mac),
+            MacAddress::UNIVERSAL_UNICAST => !self::isLocal($mac) && self::isUnicast($mac),
+            MacAddress::UNIVERSAL_MULTICAST => !self::isLocal($mac) && !self::isUnicast($mac),
+            MacAddress::UNICAST_ALL => self::isUnicast($mac),
+            MacAddress::MULTICAST_ALL => !self::isUnicast($mac),
+            MacAddress::MULTICAST_NO_BROADCAST => !self::isUnicast($mac) && !self::isBroadcast($mac),
+            MacAddress::BROADCAST => self::isBroadcast($mac),
+            MacAddress::ALL_NO_BROADCAST => !self::isBroadcast($mac),
+            default => true,
```

Comment:
why do we need this case? why `mixed` also on the signature?

Reply:
I mean the default case: can't we remove it?

Reply:
Ah I see, removed, thx!

Reply:
Update:
It's not working, because I'm not checking for all types, in this case I'm not checking for `MacAddress::ALL => !true,`, I reverted back to `default => true,`, but can also replace with `MacAddress::ALL` instead

Reply:
Yes, no default please

Reply:
done :)